core/includes/database/database.inc

```diff
@@ -671,7 +698,7 @@ abstract class DatabaseConnection {
    * query. All queries executed by Backdrop are executed as PDO prepared
    * statements.
    *
-   * @param $query
+   * @param string $query
```

Comment:
Hm, how about "An already-prepared instance of DatabaseStatementInterface may also be passed" in the comment below?

The function code considers `$query instanceof DatabaseStatementInterface`.

Reply:
Nice catch. I added that type to the docblock.